dsn.go

```diff
@@ -391,6 +401,14 @@ func parseDSNParams(cfg *Config, params string) (err error) {
 				return errors.New("invalid bool value: " + value)
 			}
 
+		// Allow fallback to unencrypted connection if server does not support TLS
+		case "allowFallbackToNoTLS":
```

Comment:
What if setting `tls=true` or verify CA and hostname with `allowFallbackToNoTLS` at the same time? What's the meaning then? Looks like invalid combination to me.

Reply:
OK, I should give more detailed description on "when will the driver fallback to no TLS", does it happen on TLS identity verify error, not supported TLS version or server not enable TLS?

I want a fallback for the last two cases, and maybe the name should contain "TLSHandshake" or something. Will learn the TLS concepts to find a good name later. Welcome to give me some advice 

also cc @dveeden